sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -622,7 +636,41 @@ public EventData(AmqpAnnotatedMessage amqpMessage)
                             long sequenceNumber = long.MinValue,
                             long offset = long.MinValue,
                             DateTimeOffset enqueuedTime = default,
-                            string partitionKey = null) : this(eventBody, properties, systemProperties, sequenceNumber, offset, enqueuedTime, partitionKey, lastPartitionSequenceNumber: null)
+                            string partitionKey = null) : this(eventBody, properties, systemProperties, sequenceNumber, offset, null, enqueuedTime, partitionKey, lastPartitionSequenceNumber: null)
```

Comment:
We shouldn't modify this one.   This exists just because we didn't have the model factory pattern back for the early releases.   This one shouldn't be used by anyone other than obsolete code.

Reply:
Removed the null but ended up needing to specify all the parameters to help the correct constructor get called.